server_test.go

```diff
@@ -712,67 +650,39 @@ func TestSubscriptionManagerSubscriptions(t *testing.T) {
 	}
 }
 
-type mockResourceLocator struct {
-	isTypeSupported func(typeURL string) bool
-	subscribe       func(typeURL, resourceName string) func()
-	resubscribe     func(typeURL, resourceName string)
-}
-
-func (m *mockResourceLocator) IsTypeSupported(_ context.Context, typeURL string) bool {
-	return m.isTypeSupported(typeURL)
-}
-
-func (m *mockResourceLocator) Subscribe(_ context.Context, typeURL, resourceName string, _ ads.RawSubscriptionHandler) func() {
-	return m.subscribe(typeURL, resourceName)
-}
+type mockResourceLocator func(typeURL, resourceName string) func()
 
-func (m *mockResourceLocator) Resubscribe(_ context.Context, typeURL, resourceName string, _ ads.RawSubscriptionHandler) {
-	m.resubscribe(typeURL, resourceName)
+func (m mockResourceLocator) Subscribe(_ context.Context, typeURL, resourceName string, _ ads.RawSubscriptionHandler) func() {
+	return m(typeURL, resourceName)
 }
 
 func TestImplicitWildcardSubscription(t *testing.T) {
 	const foo = "foo"
 	h := NewNoopBatchSubscriptionHandler(t)
 	typeURL := TypeOf[*ads.Secret]().URL()
 
-	newMockLocator := func(t *testing.T) (l *mockResourceLocator, wildcardSub, fooSub chan struct{}) {
+	newMockLocator := func(t *testing.T) (rs mockResourceLocator, wildcardSub, fooSub chan struct{}) {
```

Comment:
Nit: why is the `mockResourceLocator` renamed to `rs` instead of keeping `l`?

Reply:
Oops that's unintended